File: Data Structures & Algorithms/merge-strings-alternately/submission-0.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int i{};
        string res{""};
        while (i < word1.length() && i < word2.length()) {
            res = res + word1[i] + word2[i];
            i++;
        }
        while (i < word1.length()) {
            res+=word1[i];
            i++;
        }
        while (i < word2.length()) {
            res+=word2[i];
            i++;
        }
        return res; 
        
    }
};

File: Data Structures & Algorithms/unique-email-addresses/submission-0.py
class Solution:
    def numUniqueEmails(self, emails: List[str]) -> int:

        res = set()

        for e in emails:
            curr = ""
            after_at = False
            i = 0
            while i < len(e):
                if not after_at:
                    if e[i] != '.':
                        curr+=e[i]
                    if e[i] == '+':
                        while i < len(e):
                            if e[i] == '@':
                                break
                            i+=1 
                        after_at = True
                else:
                    curr+=e[i]
                i+=1 
            res.add(curr)
        print(res)
        return len(res)
                        


        